improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+
+    def check_threshold_coordinate_properties(self, cube_name, threshold_coord):
+        """Checks threshold coordinate properties are correct and consistent with
+        cube name"""
+        if threshold_coord.var_name != "threshold":
+            self.errors.append(
+                f"Threshold coord {threshold_coord.name()} does not have "
+                "var_name='threshold'"
+            )
+
+        self.relative_to_threshold = threshold_coord.attributes[
+            "spp__relative_to_threshold"
+        ]
+
+        if self.relative_to_threshold in ("greater_than", "greater_than_or_equal_to"):
+            threshold_attribute = "above"
+        elif self.relative_to_threshold in ("less_than", "less_than_or_equal_to"):
+            threshold_attribute = "below"
+        elif self.relative_to_threshold == "between_thresholds":
+            threshold_attribute = "between"
+            self.warnings.append("Between thresholds data are not fully supported")
+        else:
+            threshold_attribute = None
+            self.errors.append(
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}' "
+                "is not in permitted value set"
+            )
+
+        if threshold_attribute and threshold_attribute not in cube_name:
+            self.errors.append(
+                f"Cube name '{cube_name}' is not consistent with "
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}'"
+            )
+
+    def check_cell_methods(self, cube):
+        """Checks cell methods are permitted and correct"""
+        if any([substr in cube.name() for substr in PRECIP_ACCUM_NAMES]):
+            msg = f"Expected sum over time cell method for {cube.name()}"
+            if not cube.cell_methods:
+                self.errors.append(msg)
+            else:
+                found_cm = False
+                for cm in cube.cell_methods:
+                    if (
+                        cm.method == PRECIP_ACCUM_CM.method
+                        and cm.coord_names == PRECIP_ACCUM_CM.coord_names
+                    ):
+                        found_cm = True
+                if not found_cm:
+                    self.errors.append(msg)
+
+        if cube.cell_methods:
+            for cm in cube.cell_methods:
+                if cm.method in COMP_CM_METHODS:
+                    self.methods += f" {cm.method} over {cm.coord_names[0]}"
+                    if self.field_type == self.PROB:
+                        if not cm.comments or cm.comments[0] != f"of {self.diagnostic}":
+                            self.errors.append(
+                                f"Cell method {cm} on probability data should have comment "
+                                f"'of {self.diagnostic}'"
+                            )
+                    # check point and bounds on method coordinate
+                    if "time" in cm.coord_names:
```

Comment:
I don't think you check the reverse of this anywhere, i.e. if I have bounds do I have the corresponding cell method?

Reply:
We don't want that to be a blanket constraint.  For example, if I have a time cell method I should always have bounds on the time coordinate - but I may also have bounds on the forecast period coordinate, and I won't have a forecast period cell method.  So the reverse doesn't consistently apply.

Reply:
Ok, fair enough